### PR TITLE
Stratimikos Belos adapter: Handle special case for MvTimesMatAddMv

### PR DESCRIPTION
**Pull request by cgcgcg**
@trilinos/stratimikos 

## Motivation
`MvTimesMatAddMv` computes $V := \alpha AB + \beta v$ where $\alpha$ and $\beta$ are scalars, $A$ and $V$ are multivectors and $B$ is a dense matrix. 
For the case that $B$ is 1x1 we can compute $V := (\alpha B_{0,0}) A + \beta V$ instead.
This mirrors what is done for the Belos-Tpetra adapter.

### CODE REVIEW

File: packages/stratimikos/adapters/belos/src/BelosThyraAdapter.hpp [hunk 1]
```diff
@@ -290,17 +290,23 @@ namespace Belos {
...
-            0, m, 0, n,
-            arcpFromArrayView(arrayView(&B(0,0), B.stride()*B.numCols())), B.stride()
-            )
-          );
-      // perform the operation via A: mv <- alpha*A*B_thyra + beta*mv
-      Thyra::apply<ScalarType>(A, Thyra::NOTRANS, *B_thyra, Teuchos::outArg(mv), alpha, beta);
+      // Check if B is 1-by-1, in which case we can just call MvAddMv()
+      if ((m == 1) && (n == 1)) {
+        const ScalarType alphaNew = alpha * B(0, 0);
+        MvAddMv(alphaNew, A, beta, mv, mv);
```
Comment (hkthorn) [inline packages/stratimikos/adapters/belos/src/BelosThyraAdapter.hpp:296]: @cgcgcg Why not just replace this line with `Thyra::linear_combination<ScalarType>` from the `MvAddMv` method directly?

**Review by hkthorn — COMMENTED** [review]
This change makes sense, but consider maybe a direct call to a Thyra function instead of another MultiVecTraits function?

File: packages/stratimikos/adapters/belos/src/BelosThyraAdapter.hpp (hunk 1 above)
Comment (cgcgcg) [reply packages/stratimikos/adapters/belos/src/BelosThyraAdapter.hpp:296]: If that's your preference I can definitely do that!

Comment (hkthorn) [reply packages/stratimikos/adapters/belos/src/BelosThyraAdapter.hpp:296]: @cgcgcg It seems cleaner to me and removes a recursion of timers.  If you don't mind.  :)

**Comment by cgcgcg** [comment]
@sebrowne Maybe related to #12781?

**Comment by sebrowne** [comment]
@cgcgcg yes, thanks for catching that.  I think I've fixed it, I'm RETESTing (this PR will be the guinea pig for the new build)

**Comment by sebrowne** [comment]
@cgcgcg it has errors because somehow it re-enabled the tests.  I promise I checked that :( I'll disable it and try to replicate it before re-adding it.  Also affected #12784

**Comment by cgcgcg** [comment]
@sebrowne Thanks for fixing it!

**Review by hkthorn — APPROVED** [review]
LGTM

**Comment by cgcgcg** [comment]
Thanks @hkthorn !

**Pull request merged**